Value property must not be passed in to FormsyText

reebalazs opened the pull request:
Because it makes the field value unchangeable. That means that if you pass in a value to FormsyText, then since we pass ...this.props, then not only 'defaultValue', but also 'value' will be passed to TextField, and as a result, the user can never change the value of the text field.

This started to happen when I updated to material-ui 0.15.0-beta.1. 

Fixes #87.

mosanger commented:
so... what if i want to pass the value down to the text field? that's not possible anymore then.
the issue here is, that both `value` and `defaultValue` are being set to the same value. see #82

reebalazs commented:
I believe that this fixes #82, which makes #87 a duplicate.

With this PR, you can set value to FormsyText. This will pass defaultValue (but only defaultValue) to TextField, which is the correct behaviour.

mosanger commented:
it may fix your issue, but it makes it impossible to pass down a `value`

reebalazs commented:
Ok, I see. I have not thought of that.

This is the problem of the UI. One solution would be to change the UI and require to pass in defaultValue if you really want to pass in defaultValue, and value if you want value. This would be a breaking change if previously the code passed in value.

The other possibility is to apply my fix and then you loose the ability to pass in value directly.

However if everything is left as it is currently, it is not possible to set a defaultValue, and an intention to do  so will actually set value instead of defaultValue, resulting in an uneditable form.

mosanger commented:
exactly, that's why another user already proposed a fix for that. he was asked to make a PR
maybe you can try and see if it fixes your issue as well?

https://github.com/mbrookes/formsy-material-ui/issues/82#issuecomment-212508982

reebalazs commented:
Well I don't find that PR yet. Provided that the PR will do what I understand, it would fix the issue, but imo it would be a breaking change since previously I passed in value to FormsyText, and I'd need to change value to defaultValue. I could leave with that, but again, it's a breaking change.

reebalazs commented:
I am closing this PR, because the root of the problem goes deeper than this. I may consider another PR later, when I see just how deep the rabbit's hole is.
